clients/java/signalr/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -236,25 +245,27 @@ public Completable start() {
             return Completable.complete();
         }
 
-        handshakeResponseFuture = new CompletableFuture<>();
+        handshakeResponseSubject = CompletableSubject.create();
         handshakeReceived = false;
-        CompletableFuture<Void> tokenFuture = new CompletableFuture<>();
+        CompletableSubject tokenCompletable = CompletableSubject.create();
         accessTokenProvider.subscribe(token -> {
```

Comment:
This pattern is going to be a problem when we have transports other than WebSockets because we need to "re-run" the provider on each request in case the token needs to be updated...

Reply:
This should only trigger once since it's a `Single`. In the other transports we'll probably re-subscribe every time we want to call it.

Reply:
Right, but we'll need to flow it down to the transports rather than burning the token into the headers.